src/Mod/Part/App/BSplineCurvePyImp.cpp

```diff
@@ -1226,7 +1226,7 @@ PyObject* BSplineCurvePy::buildFromPolesMultsKnots(PyObject *args, PyObject *key
                 if (knots != Py_None) { number_of_knots = PyObject_Length(knots); }
                 else { //guess number of knots
                     if (Base::asBoolean(periodic)) {
-                        if (number_of_poles < degree) {degree = number_of_poles+1;}
+                        if (number_of_poles < degree) {degree = number_of_poles;}

```

Comment:
@AjinkyaDahale 
OCC does not set a constraint on degree for periodic B-Splines:
https://dev.opencascade.org/doc/refman/html/class_geom___b_spline_curve.html#a00d969dca532b7cd71594f805608fc7d

However, several algorithms fail with ```number_of_poles+1```, among them ```GCPnts_AbscissaPoint::Length```. They do not fail with ```degree = number_of_poles```.

I can only reach the conclusion that ```degree = number_of_poles + 1``` was an arbitrary limitation for a periodic B-Splines (possibly from non-periodic B-Splines). Given that algorithms fail, it could be treated as fixing a bug (although I would have welcome a more scientific rationale).

I do not foresee problems with existing projects, as what is serialised will be de-serialised to the same result.

Reply:
In theory a periodic B-spline should be able to have arbitrary degree if you're ready to repeat the poles (and knots) indefinitely.

I don't understand what you mean by "(de-)serialized".